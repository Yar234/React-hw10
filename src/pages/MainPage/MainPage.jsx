import styles from './MainPage.module.css'

export function MainPage() {

  return (
    <div className={styles.center}>
      <h1 className={styles.title}>Welcome to HOME</h1>
    </div>
  )
}