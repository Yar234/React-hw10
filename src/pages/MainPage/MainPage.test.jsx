import React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'

import { MainPage } from './MainPage'

describe('testing MainPage', () => {
  it('render page MainPage', () => {
    render(<MainPage />)
  })

  it('render text in MainPage', () => {
    render(<MainPage />)
    const El = screen.getByText(/HOME/)
    expect(El).toBeInTheDocument()
  })
  it('render button in MainPage', () => {
    render(<MainPage />)
    expect(screen.queryAllByRole('input').length).toBe(0)
  })
  it('render with snapshot', () => {
    const { asFragment } = render(<button>Change name</button>)
    expect(asFragment()).toMatchSnapshot()
  })
})