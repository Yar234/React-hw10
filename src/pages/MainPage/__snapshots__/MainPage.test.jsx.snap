// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing MainPage render with snapshot 1`] = `
<DocumentFragment>
  <button>
    Change name
  </button>
</DocumentFragment>
`;
